stellar_properties.py

```diff
@@ -1,236 +1,267 @@
-            radius, self.step_size, self.properties, auto_add=False)
-
-        self.properties['mass'].solve_differential_step(
-            radius, self.step_size, self.properties, auto_add=False)
-
-        self.properties['temperature'].solve_differential_step(
-            radius, self.step_size, self.properties, auto_add=False)
-
-        self.properties['density'].solve_differential_step(
-            radius, self.step_size, self.properties, auto_add=False)
-
-        self.properties['opticaldepth'].add_differential_step()
-        self.properties['luminosity'].add_differential_step()
-        self.properties['mass'].add_differential_step()
-        self.properties['density'].add_differential_step()
-        self.properties['temperature'].add_differential_step()
-
-        self.properties['radius'] = np.append(
-            self.properties['radius'],
-            self.properties['radius'][-1] + self.step_size)
-        self.properties["radii"] = radius
+"""
+Class defining a star and it's various differential equations.
+"""
+import numpy as np
+import desolver as de
+
+# m/s^2
+C = 2.98 * 10**8
+#m^3/kg/s^2
+G = 6.6741 * 10**-11
+# m^2*kg/s
+HBAR = 1.055 * 10**-34
+
+Me = 9.11 * 10**-31
+Mp = 1.67 * 10**-27
+#Boltzman J/K
+Kb = 1.381 * 10**-23
+# FIND THIS
+# J/k^4/m^3
+a = 7.566 * 10**-16
+
+#Star mass and radius inital guesses
+M=1.989*10**30
+R=696340000
+            
+class Star:
+    """
+    Class definining star. Can calculate many different
+    properties of the star at the various lengths across
+    it's radius
+    """
+
+    def __str__(self):
+        """
+        Print out useful information for debugging
+        """
+        info = "STAR: %s\n" % (self.name).upper()
+        info += "\n".join("Function {:21}: {}".format(key, str(value))
+                          for key, value in self.properties.items()
+                          if key not in ["energygen", "radius", "pressure"])
+
+        return info
+
+    def __init__(
+            self,
+            X=0.55,
+            Y=0.43,
+            Z=0.02,
+            Xc=0.004,
+            #Xc is the carbon mass fraction
+            cent_density=162200,
+            cent_opticaldepth=2 / 3,
+            cent_radii=0.01,  #m
+            step_size=0.1,
+            #Initial guess for total mass and radius of star based on desired position on HR diagram
+            #Define M, R above!
+            cent_temperature=2*G*M*Mp/(3*Kb*R),
+            core="Hydrogen",
+            #core is one of "Hydrogen", "Helium", "Carbon"
+            name="Generic Star"):
+        """
+        Initializes star by deffining the equations that make up
+        it's stellar structures, and their differential equations"""
+
+        self.name = name
+        self.step_size = step_size
+
+        self.cent_radii = cent_radii
+        self.cent_density = cent_density
+        self.cent_opticaldepth = cent_opticaldepth
+        self.cent_temperature = cent_temperature
+

```

Comment:
I think the issue below can be fixed by adding the line
```suggestion
self.Xc = Xc
```

Everything else seems like a great implementation.